presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PartialAggregationPushDown.java

```diff
@@ -146,31 +146,26 @@ private PlanNode pushPartial(AggregationNode partial, ExchangeNode exchange)
             for (int i = 0; i < exchange.getSources().size(); i++) {
                 PlanNode source = exchange.getSources().get(i);
 
-                if (!exchange.getOutputSymbols().equals(exchange.getInputs().get(i))) {
-                    // Add an identity projection to preserve the inputs to the aggregation, if necessary.
-                    // This allows us to avoid having to rewrite the symbols in the aggregation node below.
-                    Assignments.Builder assignments = Assignments.builder();
-                    for (int outputIndex = 0; outputIndex < exchange.getOutputSymbols().size(); outputIndex++) {
-                        Symbol output = exchange.getOutputSymbols().get(outputIndex);
-                        Symbol input = exchange.getInputs().get(i).get(outputIndex);
-                        assignments.put(output, input.toSymbolReference());
+                ImmutableMap.Builder<Symbol, Symbol> mappingsBuilder = ImmutableMap.builder();
```

Comment:
I'm not sure whether it will be so "short lived". I need it's logic for correctness here. I can not implement `PartialAggregationPushDown` without doing this symbols remapping AND this symbols remapping logic is also required in `UnaliasSymbolReferences`. Whathever happens to those classes, once both of them are moved to iterative optimizer, both will still need to share this logic.